diffkemp/simpll/DifferentialFunctionComparator.cpp

```diff
@@ -528,6 +541,14 @@ bool DifferentialFunctionComparator::maySkipInstruction(
         replacedInstructions.insert({Inst, Inst->getOperand(0)});
         return true;
     }
+    if (auto Store = dyn_cast<StoreInst>(Inst)) {
+        if (config.Patterns.StructChanges
+            && hasIgnoredAllocaAsPtrOperand(Store)) {
+            // Skip a store if it points to a previously ignored alloca.
```

Comment:
As we discussed in person, this could not lead to false negatives due to the check at the beginning of `cmpValues` which disabled value comparison of skipped allocas.

Reply:
In the updated version, where we _can_ "un-ignore" allocas, this is a problem which I fixed by converting the `ignoredInstruction` set to a map. The value in the map is a bool which specifies whether the ignored instruction can be "un-ignored" in the future. If we ever skip a store to the alloca before un-ignoring it, the value is set to false for the alloca, and from that point onward, it cannot be un-ignored.